File: 179-largest-number/largest-number.java
import java.util.Arrays;

class Solution {

    public String largestNumber(int[] nums) {
        String[] stringArray = Arrays.stream(nums)
                                     .mapToObj(String::valueOf)
                                     .toArray(String[]::new);
        
        Arrays.sort(stringArray, (s1, s2) -> {
            String tmp1 = s2 + s1;
            String tmp2 = s1 + s2;
            return tmp1.compareTo(tmp2);

        });

        String result = String.join("", stringArray);
        if (result.charAt(0) == '0') {
            return "0";
        }
        return result;
    }
}

File: 179-largest-number/largest-number.py
from functools import cmp_to_key

class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        def compare_strings(s1, s2):
            s1 = str(s1)
            s2 = str(s2)
            if s2 + s1 < s1 + s2:
                return -1
            elif s2 + s1 > s1 + s2:
                return 1
            else:
                return 0
        
        nums.sort(key=cmp_to_key(compare_strings))
        if nums[0] == 0:
            return "0"
        return "".join(map(str, nums))